h2o-core/src/main/java/water/H2O.java

```diff
@@ -367,6 +367,9 @@ private KeyValueArg(String key, String value) {
     /** -hdfs_config=hdfs_config; configuration file of the HDFS */
     public String hdfs_config = null;
 
+    /** Whole hadoop configuration to be used by H2O **/
+    public String hadoop_config = null;
```

Comment:
How is Hadoop config different from hdfs_config? It does the exact same thing. The reason why `hdfs_config` is called `hdfs` is because it configures the persistence layer (it doesn't have to be just core-site.xml - it can be anything).

Reply:
`hdfs_config` is meant to be specialized for HDFS only. For this PR, it'd be sufficient to push the whole hadoopConfig as `hdfs_config` - that's true. Having a property with the whole config to distinct a specialized `hdfs_config` from the whole hadoop configuration is just a matter if  opinion guess. I'm fine with both.

Reply:
It is unclear which one to use (for the user). And since there is no actual difference in what it does - there is no reason to confuse the users IMHO.

Reply:
Ok, I'll rework it.